### PR TITLE
[CHK-2452] add business logic to handle get state in transactions service

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Preparatory PR for handling getState NPG invocation.

#### List of Changes

Add TRANSACTION_AUTHORIZATION_REQUESTED_EVENT to dedicated queue
Update new queue refs
Define NPG notification url by setting visibility timeout for events added

#### Motivation and Context

Handling NPG notification url missing

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -508,8 +508,8 @@
                 <configuration>
                     <skipCheckoutIfExists>false</skipCheckoutIfExists>
                     <connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
-                    <scmVersionType>tag</scmVersionType>
-                    <scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+                    <scmVersionType>branch</scmVersionType>
+                    <scmVersion>CHK-2451-implements-getState</scmVersion>
```
Comment (ciuffagianluca) [inline pom.xml:512]: To be restored with stable version

File: helm/values-dev.yaml [hunk 2]
```diff
@@ -127,6 +128,7 @@ microservice-chart:
     NPG_READ_TIMEOUT: "10000"
     NPG_CONNECTION_TIMEOUT: "10000"
     NPG_CARDS_PSP_LIST: "BCITITMM,CIPBITMM,BIC36019,UNCRITMM,BPPIITRRXXX,PPAYITR1XXX"
+    NPG_AUTHORIZATION_REQUEST_TIMEOUT: "60"
```
Comment (pietro-tota) [inline helm/values-dev.yaml:131]: what about explicit this timeout measure unit in env variable name? seconds? millis?

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java [hunk 3]
```diff
@@ -4,6 +4,8 @@
 import it.pagopa.ecommerce.commons.client.NpgClient;
 import it.pagopa.ecommerce.commons.generated.npg.v1.dto.FieldsDto;
 import it.pagopa.ecommerce.commons.generated.npg.v1.dto.StateResponseDto;
+import it.pagopa.ecommerce.commons.queues.TracingUtils;
+import it.pagopa.generated.transactions.server.model.ApmAuthRequestDetailsDto;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java:8]: unused imports?

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandler.java [hunk 4]
```diff
@@ -254,11 +281,43 @@ public Mono<RequestAuthorizationResponseDto> handle(TransactionRequestAuthorizat
...
+                                                            e -> Mono
+                                                                    .just(
+                                                                            e.getData()
+                                                                                    .getPaymentGateway()
+                                                                    )
+                                                                    .filter(
+                                                                            paymentGateway -> paymentGateway
+                                                                                    .equals(PaymentGateway.NPG)
+                                                                    )
+                                                                    .switchIfEmpty(Mono.empty())
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandler.java:294]:
```suggestion
```
Filter already returns a Mono.empty() for un-matching predicate

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java [hunk 5]
```diff
@@ -404,6 +428,15 @@ void shouldSaveAuthorizationEventNpgCardsRedirectToExternalDomain() {
                 )
         ).thenReturn(Mono.empty());
 
+        Mockito.when(
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java:431]: add a verify() condition to check that `transactionAuthorizationRequestedQueueAsyncClient.sendMessageWithResponse` is effectively called once with the expected event body, TTL and visibility timeout.
This also applies also to all below junit tests.
Please add also a junit tests with other payment gateway type (such as REDIRECT) and add verifications against the fact that for payment gateways other than NPG this event is not written in the queue

**Comment by pietro-tota** [comment]
please take a look at CR integration tests, they seems to be broken

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java (hunk 5 above)
Comment (ciuffagianluca) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java:441]: Handled in 76aa9d14

File: helm/values-uat.yaml [hunk 6]
```diff
@@ -118,6 +118,7 @@ microservice-chart:
     TRANSACTION_CLOSE_PAYMENT_QUEUE_NAME: pagopa-u-weu-ecommerce-transactions-close-payment-queue
     TRANSACTION_REFUND_QUEUE_NAME: pagopa-u-weu-ecommerce-transactions-refund-queue
     TRANSACTION_NOTIFICATIONS_QUEUE_NAME: pagopa-u-weu-ecommerce-transaction-notifications-queue
+    TRANSACTIONS_AUTHORIZATION_REQUESTED_QUEUE_NAME: pagopa-d-weu-ecommerce-transaction-auth-requested-queue
```
Comment (pietro-tota) [inline helm/values-uat.yaml:121]: please add parameter in canary delivery parameters with -b suffix, as done [here](https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/0d4e622f9764bc2b5a91789f202095f40374e6fe/helm/values-uat.yaml#L25)

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import it.pagopa.ecommerce.commons.client.NpgClient;
 import it.pagopa.ecommerce.commons.generated.npg.v1.dto.FieldsDto;
 import it.pagopa.ecommerce.commons.generated.npg.v1.dto.StateResponseDto;
+import it.pagopa.generated.transactions.server.model.ApmAuthRequestDetailsDto;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java:7]: unused import?

**Review by infantesimone — APPROVED** [review]

**Pull request merged**